xbmc/addons/AddonInstaller.cpp

```diff
@@ -675,7 +675,7 @@ bool CAddonInstallJob::DoWork()
   {
     origin = ORIGIN_SYSTEM; // keep system add-on origin as ORIGIN_SYSTEM
   }
-  else if (m_addon->HasType(ADDON_REPOSITORY))
+  else if (m_addon->IsMainType(ADDON_REPOSITORY))
```

Comment:
and i am wondering if the fix for this crash is a simple
`else if (m_addon->MainType() == ADDON_REPOSITORY)`
but this isn't tested

Reply:
For there becomes only the first entry in addon.xml with `<extension point="...">` (xbmc.addon.repository) used.

All other parts and later "extension point"'s becomes ignored.
The `HasType` looks also about subparts:
```xml
    <extension point="xbmc.python.pluginsource" library="resources/lib/default.py">
        <provides>video</provides>
    </extension>
```
On this should `ADDON_PLUGIN` and `ADDON_VIDEO` true.
And all other `extension point` parts becomes checked about type.

Reply:
As a result, the difference between CAddon and CAddonInfo has to be taken into account.

- CAddonInfo gives the complete addon with all its parts
- CAddon brings the respective part to use in it

As a result, a CAddonInfo can e.g. belong to 5 different CAddon classes.